_posts/2024-02-17-social-integration.md

```diff
@@ -0,0 +1,126 @@
+We have been doing that for years and years now, via our [BaNaNaS system](https://bananas.openttd.org).
+This delivers in-game content to the user, while the author of such content can license it how ever they like.
+The only "demand" we make, is that the author gives us the right to distribute their content, and that is is available free of charge.
+
+In result, we see a wide variety of content on BaNaNaS, under all kinds of licenses.
+Some authors embrace the ideology of OpenTTD, and use open or free licenses.
+Others are more protective of their work, and do not allow anyone to make derivatives of their work.
+From our point of view, we don't actually mind either way: as long as our players can download it for free and enjoy the content, we are more than happy.
+
+The plugin system that is added to 14.0 is not much different from this.
+Although technically it is very different, more on that later, but from your point of view it is the same: download a plugin, put it in the right folder, and enjoy the added functionality.
+
+Currently we release three plugins: Steam, Discord, and GOG Galaxy.
+The source of those plugins are released under MIT licenses; another Open Source license, with much less conditions compared to GPLv2.
+
+## Security is a thing
+
+Content uploaded to BaNaNaS is executed in the game via sandboxes: nobody can read files on your computer, make network connections, or anything like that.
+They can only do things we allow them, which is a very restrictive set of things.
+
+This is a lot harder for a plugin system that integrates with platforms like Steam, Discord, GOG Galaxy, etc.
+We can't sandbox those really, as they have to be actual binaries that run on the player's system.
+
+As such, we have a dilemma: if we would allow a plugin system in the same way we allow BaNaNaS content, anyone could upload a plugin for anyone to download.
+And that plugin could, in theory, contain things that are harmful to the player.
+For example a bitcoin miner or something silly.
+
+This meant we had to find a balance between allowing such plugins and not allowing anyone to just provide such plugins.
+What we came up with, is a system where a plugin can only be loaded into OpenTTD, if we, The OpenTTD Development Team, approved the plugin.
+
+We find this less than ideal, as we like that for example BaNaNaS allows anyone to come up with ideas, try them out, see what people think, etc.
+But, we also have to keep the security of players in mind.
+And there is just too much potential for harm when we would open up the plugin system to anyone.
+As such, we were left with no other choice than to restrict who can make those plugins; or of course, not do it at all.
+
+In result, every plugin is released and signed by OpenTTD, and before loaded into the game, it is validated the signature is correct.
+If you are a developer and want to design your own plugin, please come talk to us.
```

Comment:
So a user simply can't develop a plugin of their own and make it load?

Reply:
Not in a release version of the game, no. Ofc for developers there are ways around there, but this blog is not for them :)

Reply:
That's the more reason IMO to also make a (possibly in-box for all distributions) plugin that simply writes the status to a JSON file in the config folder, or perhaps a filename controlled by an environment variable (if set), to let users get access to the data in a simple way without needing a specific plugin. It will also fulfill the role of a plugin that does not need to link with non-GPL-compatible libraries.

Reply:
As always, contributions are welcome :) Either way, not for this PR :)